src/pages/Home/components/movie-modal-form/MovieModalForm.tsx

```diff
@@ -1,104 +1,170 @@
+            vote_average: formValues?.vote_average || '',
+            vote_count: formValues?.vote_count || '',
+            release_date: formValues?.release_date || '',
+            poster_path:  formValues?.poster_path || '',
+            overview:  formValues?.overview || '',
+            budget:  formValues?.budget || '',
+            revenue:  formValues?.revenue || '',
+            runtime:  formValues?.runtime || '',
+            genres:  formValues?.genres[0] || ''
+        } as IMovieItem,
+        validate: FORM_VALIDATE,
+        onSubmit: async (values, {resetForm}) => {
+            const movie: IMovieItem = setEmptyMovieFields(values)
+            let responseStatus
+            if (formValues) {
+                const updatedMovie = {
+                    ...movie,
+                    id: formValues.id
+                }
+                responseStatus = await updateMovie(updatedMovie)
+            } else {
+                responseStatus = await addMovie(movie)
+            }
+            if (+responseStatus <= 299) {
+                resetForm()
+                fetchMovies(filterBy, sortBy)
+                setAddMovieSuccess(true)
+            }
+        }
+    })
 
     return (
-        <Modal>
-            <form
-                className={styles.addMovieForm}
-                onSubmit={addMovieHandler}
-            >
-                <p className={classnames('h1', styles.formTitle)}>{modalName}</p>
-                <div className={classnames('row', 'smaller', styles.formGroup)}>
-                    <div className={classnames('col', 'smaller', 'col-8')}>
-                        <Input
-                            inputType={'text'}
-                            inputName={'movieTitle'}
-                            labelText={'Title'}
-                            inputPlaceholder={'Select Title'}
-                            inputValue={formValues?.title}
-                        />
+        <Modal>{
+            !addMovieSuccess ? (
+                <form
+                    className={styles.addMovieForm}
+                    onSubmit={movieForm.handleSubmit}
+                >
+                    <p className={classnames('h1', styles.formTitle)}>{modalName}</p>
+                    <div className={classnames('row', 'smaller', styles.formGroup)}>
+                        <div className={classnames('col', 'smaller', 'col-8')}>
+                            <Input
+                                inputType={'text'}
+                                inputName={'title'}
+                                labelText={'Title'}
+                                inputPlaceholder={'Select Title'}
+                                onChange={movieForm.handleChange}
+                                onBlur={movieForm.handleBlur}
+                                inputValue={movieForm.values.title}
+                                errorMessage={movieForm.touched.title ? movieForm.errors.title : null}
```

Comment:
Do we need check `touched` state for displaying error? As I see it should work correctly as is.

Reply:
Not exactly, If we miss touched require error will show for all required fields when focusout action will do for one field. I checked it.